### ISSUE TITLE
fix Reading error on newer versions of Shapely

### ISSUE THREAD

**Issue by tomkralidis** [geopython/pycsw#584]
# Description

From @jmckenna

> The only change I had to make was to use "WKTReadingError" in
> pycsw/core/repository.py (line#41), because I am using a much newer
> version of Shapely and GEOS.  For the record, I couldn't find anyone
> else reporting that change.

# Environment

- operating system: all
- Python version: 3.7
- pycsw version: master
- source/distribution
  - [x] git clone
  - [ ] DebianGIS/UbuntuGIS
  - [ ] PyPI
  - [ ] zip/tar.gz
  - [ ] other (please specify):
- web server
  - [ ] Apache/mod_wsgi
  - [ ] CGI
  - [ ] other (please specify): 

# Steps to Reproduce

# Additional Information

https://lists.osgeo.org/pipermail/pycsw-devel/2019-January/000805.html

**Comment by jmckenna** [comment]
Here was the exact error thrown by pycsw (important is the last message of "ImportError: cannot import name 'ReadingError' from 'shapely.geos" ):

```
Traceback (most recent call last):\r: C:/ms4w/apps/pycsw-git-master/csw.py
  File "C:\\ms4w\\Python\\lib\\wsgiref\\handlers.py", line 137, in run\r: C:/ms4w/apps/pycsw-git-master/csw.py
    self.result = application(self.environ, self.start_response)\r: C:/ms4w/apps/pycsw-git-master/csw.py
  File "C:\\ms4w\\apps\\pycsw-git-master\\pycsw\\wsgi.py", line 77, in application\r: C:/ms4w/apps/pycsw-git-master/csw.py
    status, contents = csw.dispatch_wsgi()\r: C:/ms4w/apps/pycsw-git-master/csw.py
  File "C:\\ms4w\\apps\\pycsw-git-master\\pycsw\\server.py", line 250, in dispatch_wsgi\r: C:/ms4w/apps/pycsw-git-master/csw.py
    return self.dispatch()\r: C:/ms4w/apps/pycsw-git-master/csw.py
  File "C:\\ms4w\\apps\\pycsw-git-master\\pycsw\\server.py", line 398, in dispatch\r: C:/ms4w/apps/pycsw-git-master/csw.py
    from pycsw.core import repository\r: C:/ms4w/apps/pycsw-git-master/csw.py
  File "C:\\ms4w\\apps\\pycsw-git-master\\pycsw\\core\\repository.py", line 41, in <module>\r: C:/ms4w/apps/pycsw-git-master/csw.py
    from shapely.geos import ReadingError\r: C:/ms4w/apps/pycsw-git-master/csw.py
ImportError: cannot import name 'ReadingError' from 'shapely.geos' (C:\\ms4w\\Python\\lib\\site-packages\\shapely\\geos.py)\r: C:/ms4w/apps/pycsw-git-master/csw.py

```

**Comment by jmckenna** [comment]
Also note that this is with the last Shapely stable release: Shapely==1.6.4.post2  (see https://pypi.org/project/Shapely/)

**Comment by tomkralidis** [comment]
Thanks much @jmckenna for the detailed report.  Fixed in #586.  Any chance you can validate?

Having said this, @sgillies any idea when this changed in Shapely, so we can point to any release notes for reference?

**Comment by sgillies** [comment]
2 years ago. The root of the problem, @tomkralidis, is that pycsw is too specific about the exception handling. Shapely's WKTReadingError derives from ReadingError, so at https://github.com/geopython/pycsw/blob/master/pycsw/core/repository.py#L489 you should just handle ReadingError. Also: https://github.com/geopython/pycsw/blob/master/pycsw/core/repository.py#L41-L44 is incorrect, you should do `from shapely.errors import ReadingError`.

**Comment by jmckenna** [comment]
Thanks @tomkralidis and @sgillies 

I waited to test with yesterday's Apache HTTP release.  I confirm this fix, with today's pycsw-master, Shapely 1.6.4.post2 and Apache 2.4.38 through mod_wsgi, on Windows.

**Comment by tomkralidis** [comment]
Thanks @jmckenna
